### ISSUE TITLE
Rustdesk 1.2 Nightly doesnt work with "map mode" on Debian lxqt or Lubuntu with lxqt

### ISSUE THREAD

**Issue by Darkyere** [rustdesk/rustdesk#2829]
**Describe the bug you encountered:**

When connecting with the newest Nightly of rustdesk (currently 1.2) from MAC to Debian with lxqt the stream crashes as soon as a button on the keyboard is pressed.
This is the same if choosing Legacy Mode.

I origionally thought it might be a problem with my Debian lxqt setup.

So i booted up Lubuntu 22.04 with lxqt and when connecting from MAC to Lubuntu 22.04 it does not crash the stream but the map mode does not work.
It instead sends the "US" keys instead of the danish.
Legacy mode also only sends "US" version of keys but that makes sense.

->

On the other hand when going from Windows to Debian lxqt
The Map mode does not crash, but it sends the "US" keys.
With legacy mode the stream crashes.

On windows to Lubuntu 22.04 both Map mode and Legacy mode sends the US keyboad but the stream does not crash.

...

**What did you expect to happen instead?**

That the stream does not crash and that Map mode works with DK layout.

...

**How did you install `RustDesk`?**

Rustdesk Github release page, nightly build.

---

**RustDesk version and environment**

From Win\MAC

To lxqt desktop environment (only tested Debian with lxqt and Lubuntu 22.04)

I have Win/MAC set to Danish OS including Danish keyboard mapping.
Windows makes use of "Windows" keyboard.
MAC makes use of "MAC" keyboard.

I have Debian lxqt and Lubuntu set to English OS but with Danish keyboard layout.

----

Looking forward for the final release,
Thank you for reading.

Best regards
Darkyere

**Comment by rustdesk** [comment]
#1547 @fufesou

**Comment by Darkyere** [comment]
Sry. i was maybe missing some info.

I have been helping testing out the ARM64 version for Ubuntu RPI4 and the MAC version for Danish keyboard input and on those the Map mode works.

[Raspberry PI 4 with Ubuntu ARM64](https://github.com/rustdesk/rustdesk/issues/853)
[Keyboard mapping not working for DK layout](https://github.com/rustdesk/rustdesk/issues/994)

Not because i have done anything else than asking for help and testing they worked afterwords for  Danish keyboard/Map mode.

I have also when i was at it tested Rustdesk for Windows and Ubuntu GNOME/KDE and these work as well for danish keyboard/Map mode.

Basically i have tested every configuration between windows/mac/rpi4/"ubuntu GNOME+KDE" and they all work with danish keyboard and map mode.

This seems to be related to lxqt on both Debian and Ubuntu since all other tests "i" have been able to make has worked so far.

Hope you can have a look at it since ive never tried an entire stream crashing just by pressing a normal letter on the keyboard before and only on lxqt :)

Id be happy to test any executables if this issue can be looked into.

Best regards
Darkyere

**Comment by rustdesk** [comment]
Sorry, we have no resource to test or support lxqt.

**Comment by Darkyere** [comment]
That is understandable.
I still appreciate the work and look forward for the final release of 1.2.

If it should happen that someone at some point could look at it i would be happy to help test a release from the nightly channel :)

**Comment by rustdesk** [comment]
@fufesou this is marked as "help wanted", we should not put time on this in the near future before we finish the other milestones.

**Comment by grummbeer** [comment]
@Darkyere 

> the map mode does not work.
> It instead sends the "US" keys instead of the danish.

i have a debian/lxqt VM to test rustdesk and experienced the same issues. But with german layout. All connections to debian/lxqt, from other linux or window behave exactly this … but then … it works.

Do you have reboot your debian/lxqt and then doublecheck the keyboard settings in lxqt?

For the full story start here https://github.com/rustdesk/rustdesk/issues/406#issuecomment-1380664128.

**Comment by Darkyere** [comment]
Yes the problem seems to be conistent.

Its a PC (Debian lxqt) and a Mini PC (Lubuntu 22.04)
Its devices i turn on when i need to use them.

----

I first experienced it while being at home from Win/Mac to lxqt.
Used x11vnc instead and shut down when i was done.

A few days later i was at my old folks, so i powered them on remotely and tried accessing them with the Windows PC they own and my Arch Linux laptop i have in their home.
This was when it hit me it was happening on other places than my home.

Shut down the devices with x11vnc again.

----

So the next time i was home is when i started making my from Win/Mac/Linux to lxqt test and made the origionally post here.
The devices has been rebooted multiple times duing the test.

But it is a little more than just map mode doesnt work and it uses "US" keyboard instead.

On Debian lxqt, the stream entirely crashes with the message "Reset by peer" just by pressing any letter, including something as simple as an "a".

----

I understand there is not resources to look at it for now.
But i dont mind testing nightlys when there is time to look at it.

I am glad to see you stumbled on my issue and shared your own experince with me.
It's allways good to see there are other situations that are similar to my own.

Maybe solving "one" will solve the "other" 👍
